### PR TITLE
chore: adding flex and margin style props

### PR DESCRIPTION
**Pull request by zchenwei**
*Description of changes:*

1. Adding flex and margin style props
2. Making all props required in `ComponentPropToStyleProp` so that any missing style props in `ComponentPropsToStylePropsMap` will be caught.
3. Adding interface to type `useAmplify` hook explicitly

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/react/src/primitives/types/style.ts [hunk 1]
```diff
@@ -53,6 +53,13 @@ export interface BaseStyleProps extends FlexItemStyleProps, GridItemStyleProps {
   height?: ResponsiveStyle<StyleToken<Property.Height>>;
   left?: ResponsiveStyle<StyleToken<Property.Left>>;
   letterSpacing?: ResponsiveStyle<Property.LetterSpacing>;
+  lineHeight?: ResponsiveStyle<Property.LineHeight>;
+  margin?: ResponsiveStyle<Property.Margin>;
+  maxHeight?: ResponsiveStyle<Property.MaxHeight>;
+  maxWidth?: ResponsiveStyle<Property.MaxWidth>;
+  minHeight?: ResponsiveStyle<Property.MinHeight>;
+  minWidth?: ResponsiveStyle<Property.MinWidth>;
+  opacity?: ResponsiveStyle<Property.Opacity>;
```
Comment (dbanksdesign) [inline packages/react/src/primitives/types/style.ts:62]: Could we change these lines to `ResponsiveStyle<StyleToken<Property.*>>;` ?

**Review by dbanksdesign — APPROVED** [review]
Not blocking but we could have the new style props accept design tokens. :shipit:

File: packages/react/src/primitives/types/style.ts (hunk 1 above)
Comment (zchenwei) [reply packages/react/src/primitives/types/style.ts:62]: Oh yea, I would be happy to do that

**Review by reesscot — APPROVED** [review]
🚢!

**Pull request merged**